qcodes/instrument/parameter.py

```diff
@@ -609,7 +559,7 @@ def get_wrapper(*args: Any, **kwargs: Any) -> ParamDataType:
                 if self._validate_on_get:
                     self.validate(value)
 
-                self._update_cache_with(value=value, raw_value=raw_value)
+                self.cache._update_with(value=value, raw_value=raw_value)
```

Comment:
(discussed offline) adding such a kwarg is not nice because it interconnects various methods which makes code less clean. `_save_val` had such a kwarg, and hence was too powerful and also not really at the same time. So, offline it was agreed to not do this.